hazelcast/src/main/java/com/hazelcast/client/impl/clientside/ClientDynamicClusterConfig.java

```diff
@@ -134,6 +135,9 @@ public Config addMapConfig(MapConfig mapConfig) {
         Data partitioningStrategy = mapConfig.getPartitioningStrategyConfig() == null
                 ? null : serializationService.toData(mapConfig.getPartitioningStrategyConfig().getPartitioningStrategy());
 
+        DataPersistenceAndHotRestartMerger
+                .merge(mapConfig.getHotRestartConfig(), mapConfig.getDataPersistenceConfig());
```

Comment:
Question: Do we need to send the `DataPersistenceConfig` element in the `DynamicConfigAddMapConfigCodec` request below? I see that we only do it for `HotRestartConfig`. Or, merging the configuration elements and then just sending the `HotRestartConfig` is enough?

Reply:
It is an issue, however unrelated to this PR. See also https://github.com/hazelcast/hazelcast-enterprise/issues/4282